wagtail/images/forms.py

```diff
@@ -37,6 +38,29 @@ def formfield_for_dbfield(db_field, **kwargs):
 class BaseImageForm(BaseCollectionMemberForm):
     permission_policy = images_permission_policy
 
+    def __init__(self, *args, **kwargs):
+        super().__init__(*args, **kwargs)
+        self.original_file = self.instance.file
+
+    def save(self, commit=True):
+        if "file" in self.changed_data:
+            self.instance._set_image_file_metadata()
+
+        super().save(commit=commit)
+
+        if commit:
+            if "file" in self.changed_data and self.original_file:
+                # if providing a new image file, delete the old one and all renditions.
+                # NB Doing this via original_file.delete() clears the file field,
+                # which definitely isn't what we want...
+                self.original_file.storage.delete(self.original_file.name)
+                self.instance.renditions.all().delete()
```

Comment:
Would removing renditions be more appropriate in a post_save or post_delete signal so they run on API or direct ORM operations as well?